### PR TITLE
BS-13862: Don't deploy Business Object Definitions when application is

### PR DESCRIPTION
**Pull request by apupier**
in headless mode (Tomcat with the engine is not started)

### CODE REVIEW

File: bundles/plugins/org.bonitasoft.studio.businessobject/src/org/bonitasoft/studio/businessobject/core/operation/DeployBDMOperation.java [hunk 1]
```diff
@@ -70,6 +71,7 @@ public void setSession(final APISession session) {
      */
     @Override
     public void run(final IProgressMonitor monitor) throws InvocationTargetException, InterruptedException {
+        assert (!PlatformUtil.isHeadless());
```
Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.businessobject/src/org/bonitasoft/studio/businessobject/core/operation/DeployBDMOperation.java]: Becarefull with those assert. They will not fail if jvm is not launched with specific argument (-ea).
You may use a Preconditions.checkState() instead.

Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.businessobject/src/org/bonitasoft/studio/businessobject/core/operation/DeployBDMOperation.java]: we are not launching with it by default?

Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.businessobject/src/org/bonitasoft/studio/businessobject/core/operation/DeployBDMOperation.java]: i don't think so and I don't think you want it toggable here.

**Comment by rbioteau** [comment]
I think the logic is missplaced
You should put it in the DeployBDMOperation.
We will want, even in headless, the bdm jar to be created. It will not coded like that.

**Comment by apupier** [comment]
Not agree, if I call DeployBDMOperation, I want it to be deployed and not to be ignored in case I'm in headless mode

**Comment by rbioteau** [comment]
Forgot what I said. The jar comes from the engine.
So we may encounter dome issues in headless build if bdm jar is not committed.

**Pull request merged**